stablehlo/conversions/linalg/transforms/StablehloLegalizeToLinalg.cpp

```diff
@@ -419,29 +419,49 @@ struct BroadcastConverter final
   }
 };
 
+template <typename OpTy>
+int64_t getBroadcastSizes(OpTy op) {
+  if constexpr (std::is_same<OpTy, BroadcastOp>::value)
+    return op.getBroadcastSizes().size();
+  if constexpr (std::is_same<OpTy, BroadcastInDimOp>::value)
+    return op.getType().getRank() - op.getBroadcastDimensions().size();
```

Comment:
Let's just define this method twice? Or use:

```
typename std::enable_if<std::is_same<OpTy, BroadcastOp>::value ||
                            std::is_same<OpTy, BroadcastInDimOp>::value,
                            int64_t>::type;
```

Let's keep this as a compiler error instead of a runtime error

Reply:
I ended up just rewriting this function with 2 overloads. I think I went for the template solution initially out of cognitive availability from having just done some template stuff, but overloading is simpler in this case.